include/dkm_matrix.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#ifndef DKM_MATRIX_HPP
+#define DKM_MATRIX_HPP
+
+#include <vector>
+#include <cassert>
+#include <algorithm>
+
+namespace dkm
+{
+// This class is only an interface! Not designed to be used outside library internals.
+
+template<typename T>
+class as_matrix
```

Comment:
I think it would be fine to just name this `matrix`.

Reply:
The name comes from the fact that we are dealing with the data "as a matrix". But `as_matrix` indeed is not even a matrix, it is only an interface between the algorithm and the data. I think the name `as_matrix` express better the idea.